recipes/libmeshb/all/conanfile.py

```diff
@@ -0,0 +1,90 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, replace_in_file, rm, rmdir
+from conan.tools.microsoft import check_min_vs, is_msvc, is_msvc_static_runtime
+from conan.tools.scm import Version
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class LibmeshbConan(ConanFile):
+    name = "libmeshb"
+    description = "A library to handle the *.meshb file format."
+    license = "MPL-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/LoicMarechal/libMeshb"
+    topics = ("3d", "mesh", "geometry")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
+        
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def validate(self):
+        if is_msvc(self) and self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared on Visual Studio and msvc.")
```

Comment:
A comment on the reason why would be useful here. I'm assuming the symbols are not exported?

Reply:
Hi @valgur,

I don't know very well C++ specific on Windows.
If I understand correctly to do shared lib on Windows code must include export macro (https://learn.microsoft.com/en-us/cpp/build/reference/export-exports-a-function?view=msvc-170)?
When looking at https://github.com/LoicMarechal/libMeshb/blob/master/sources/libmeshb7.h I don't see such export macro.
It seems that this can be managed by CMake (https://www.kitware.com/create-dlls-on-windows-without-declspec-using-new-cmake-export-all-feature/).

I have removed this valide method but now CI fails on Windows with shared option at true.

I'll try "tc.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True" in generate() method to fix that.

Reply:
> I'll try "tc.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True" in generate() method to fix that.

Please, don't. We try to avoid adding feature not supported by the upstream. Plus, `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS` is fragile, it exposes all symbols, including private and is not capable to export static variables, causing linkage errors. In the upstream does not support shared on Windows, neither do we.